src/base/server_type.cpp

```diff
@@ -271,6 +272,63 @@ bool is_tran_server_with_remote_storage ()
   return false;
 }
 
+/*
+ * get_page_server_maxim_extra_thread_count - returns the number of extra thread count per
+ *        scalability server type
+ *
+ */
+int get_maxim_extra_thread_count_by_server_type ()
+{
+  SERVER_TYPE local_server_type = g_server_type;
+  if (local_server_type == SERVER_TYPE_UNKNOWN)
+    {
+      // by this time, the server type is in one of two states:
+      //  - either non initialized, in which case it will be initialized later from config file
+      //  - or, explicitly initialized from command line
+      // in the case of the first state, we need the config file value here
+      // NOTE: further explanation, we cannot move the initialization of the entire "server type" infrastructure -
+      //  - aka, the function 'init_server_type' - before thread infrastructure initialization because server
+      // type initialization already initializes some threads and, on the other hand, the threading infrastructure
+      // needs this function to properly initialize itself
+      const auto server_type_from_config = (server_type_config) prm_get_integer_value (PRM_ID_SERVER_TYPE);
+      local_server_type = get_server_type_from_config (server_type_from_config);
+    }
+
+  if (local_server_type == SERVER_TYPE_PAGE)
+    {
+      // thread pool used by page server to perform parallel replication
+      //
+      int replication_parallel_thread_count_min = -1;
+      int replication_parallel_thread_count_max = -1;
+      int prm_error = sysprm_get_range (prm_get_name (PRM_ID_REPLICATION_PARALLEL_COUNT),
+					&replication_parallel_thread_count_min, &replication_parallel_thread_count_max);
+      if (prm_error != PRM_ERR_NO_ERROR)
+	{
+	  // a safe bet
+	  replication_parallel_thread_count_max = std::thread::hardware_concurrency ();
+	}
+
+      // thread pool used by page server to asynchronously service requests from transaction servers
+      //
+      int server_request_responder_thread_count_min = -1;
+      int server_request_responder_thread_count_max = -1;
+      prm_error = sysprm_get_range (prm_get_name (PRM_ID_SCAL_PERF_PS_REQ_RESPONDER_THREAD_COUNT),
+				    &server_request_responder_thread_count_min, &server_request_responder_thread_count_max);
+      if (prm_error != PRM_ERR_NO_ERROR)
+	{
+	  // a safe bet
+	  server_request_responder_thread_count_max = std::thread::hardware_concurrency ();
+	}
+
+      return replication_parallel_thread_count_max + server_request_responder_thread_count_max;
```

Comment:
Since we have two responder though it will be integrated into one in the end.
```suggestion
      return replication_parallel_thread_count_max + server_request_responder_thread_count_max * 2;
```

Reply:
Done.